fml/task_runner.cc

```diff
@@ -23,6 +24,15 @@ void TaskRunner::PostTask(fml::closure task) {
   loop_->PostTask(std::move(task), fml::TimePoint::Now());
 }
 
+void TaskRunner::AwaitTask(fml::closure task) {
+  CountDownLatch latch(1);
```

Comment:
This will deadlock if the caller is already on the specified task runner. This happens on the test runner where all task runner references are to the same underlying task runner (making the test runner effectively single threaded). Even after the deadlock is fixed, this seem like a too easy of method to abuse and I would rather have the caller implement the same.

Reply:
The authors of grand central dispatch disagree "dispatch_sync" also Scala's actors has synchronous message sending `!?`.  I think the concern isn't giving the authors enough credit.  I've been thinking about this though and I don't think I need to sync on this technically.

Reply:
I've removed this for now since I don't believe it is necessary despite it being easier to reason about if it is synchronous imo.